core/parser.cpp

```diff
@@ -1186,9 +1186,42 @@ namespace {
 
                     Token op = pop();
                     if (op.kind == Token::BRACKET_L) {
-                        AST *index = parse(MAX_PRECEDENCE);
+                        const int max = 2;
+                        bool isSlice = true;
```

Comment:
The coding convention here is that locals on the stack follow the C style of foo_bar whereas fields of classes use fooBar.  While I'm at it, macros and constants are FOO_BAR.